### PR TITLE
MSCX class API for creating score excerpts

### PR DESCRIPTION
**Pull request by leobruneau**
1) make_excerpt() : takes starting and ending (optional) measures and then creates an excerpt after retrieving corresponding globalkey and localkey information
2) extract_phrases() : calls make_excerpt() method for each pair of starting and ending measure corresponding to a phrase
3) extract_random_snippets() : takes as parameters number of snippets and snippet length to randomly create excerpts (calls the make_excerpt() method for each randomly generated pair)

### CODE REVIEW

File: src/ms3/score.py [hunk 1]
```diff
@@ -1081,6 +1086,200 @@ def _update_annotations(self, infer_types={}):
...
+        row = get_row_at_quarterbeat(df=self.expanded(), quarterbeat=quarterbeat_start)
+
+        # TODO: Check if this is correct (sometimes get_row_at_quarterbeat returns a DataFrame instead of a Series)
+        if isinstance(row, pd.DataFrame):
+            row = row.iloc[-1]
+
+        global_key = row["globalkey"]
+        local_key = row["localkey"]
+
+        print(f"Global key: {global_key}, Local key: {local_key}")
```
Comment (johentsch) [inline src/ms3/score.py:1192]: please use `self.logger.debug`

**Review by johentsch — CHANGES_REQUESTED** [review]

**Comment by leobruneau** [comment]
Requested changes were met

**Review by johentsch — CHANGES_REQUESTED** [review]
Hi @leobruneau,

Please go through my added commits carefully (you can [start her](https://github.com/johentsch/ms3/pull/91/commits/c27e7cdcf14a41f7bca6dd428dca787382fbf867) and conveniently use the arrow button for that) to see if you agree with all the changes. Feel free to leave comments, questions, suggestions.

When you're through, please add your name to the contributors list in `AUTHORS.rst`. Then we might be ready to merge, depending on the outcome of your review.

**Review by leobruneau — COMMENTED** [review]
You made some very nice changes. I'm sorry that you had to re-factor almost everything. I'm not sure I should be added to the contributors.

I think it's all ready to merge

**Review by johentsch — APPROVED** [review]
Thank you, Léo!

**Pull request merged**